File: TDD Boot Camp/Part 01/Code/OOPrinciples/src/com/oop/abstraction/Elan.java
package com.oop.abstraction;


import java.time.LocalDate;
import java.time.Duration;
public class Elan {
	public String firstName;
	public String lastName;
	public LocalDate  dateOfBirth;

	public Elan() {
		this.firstName = "Elan";
		this.lastName = "Sreenivasan";
		this.dateOfBirth = LocalDate.of(2018, 11, 13);
	}
	
	public int getAge() {
		return calculateAge();
	}
	
	public int calculateAge() {
		long duration =  Duration.between(LocalDate.now(), this.dateOfBirth).getSeconds();
		return (int)(duration/(60*60*24*365));
	}
}

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Elan.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Abstraction
{
    public class Elan
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }

        public Elan()
        {
            FirstName = "Elan";
            LastName = "Sreenivasan";
            DateOfBirth = new DateTime(2018, 11, 13);
        }

        public int GetAge()
        {
            return CalculateAge();
        }

        public int CalculateAge()
        {
            double duration = (DateTime.Now - DateOfBirth).TotalSeconds;
            return (int)(duration / (60 * 60 * 24 * 365));
        }
    }
}
